### PR TITLE
RDM-6320 start case creation as case worker-BE-FAT

### PR DESCRIPTION
**Pull request by shrivsksavit**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-6320

### Change description ###

Implement Functional Tests for 'CCD Data Store API:: Start case creation as Caseworker' operation

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Comment by shrivsksavit** [comment]
https://github.com/hmcts/ccd-data-store-api/pull/641

**Comment by shrivsksavit** [comment]
https://github.com/hmcts/ccd-data-store-api/pull/655/files

File: src/aat/resources/features/F-049/F-049.feature [hunk 1]
```diff
@@ -0,0 +1,114 @@
+@F-049
+Feature: Start case creation as Case worker
+
+  Background: Load test data for the scenario
+    Given an appropriate test context as detailed in the test data source
+
+  @S-235
+  Scenario: must start case creation process successfully for correct inputs
+    Given   a user with [a detailed profile in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: with an active profile in CCD

File: src/aat/resources/features/F-049/F-049.feature [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-401]
+    And     the response has all the details as expected
+
+  @S-231
+  Scenario: must return 403 when request provide invalid authentication credentials
+    Given   a user with [a detailed profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [provide invalid authentication credentials]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: the request [provides a dummy user id]

File: src/aat/resources/features/F-049/S-232.td.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "_guid_": "S-232",
+  "_extends_": "F-049_Test_Data_Base",
+  "title": "must return 403 when request provide unauthorized access",
+  "specs": [
+    "a detailed profile in CCD",
+    "Start case creation as Case worker",
+    "CCD Data Store",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-232.td.json]: What are these?

File: src/aat/resources/features/F-049/S-232.td.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    "provide unauthorized access",
+    "code is HTTP-403"
+  ],
+  "request": {
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-232.td.json]: where is the unauthorized access in this request?

File: src/aat/resources/features/F-049/F-049.feature [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-404]
+    And     the response has all the details as expected
+
+  @S-511
+  Scenario: must return 403 when request provide a jurisdiction id not existing in ccd
+    Given   a user with [a detailed profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [provide a jurisdiction id not existing in ccd]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: Grammar please please please

File: src/aat/resources/features/F-049/F-049.feature [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-403]
+    And     the response has all the details as expected
+
+  @S-234 @Ignore
+  Scenario: must return 400 when request provide special character for Case type ID in CCD
+    Given   a user with [a detailed profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [provide special character for Case type ID in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: Grammar please please please

File: src/aat/resources/features/F-049/F-049.feature [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    And     it is submitted to call the [Start case creation as Case worker] operation of [CCD Data Store]
+    Then    a negative response is received
+    And     the response [code is HTTP-403]
+    And     the response has all the details as expected
+
+  @S-512
+  Scenario: must return 404 when request provide an invalid Case type ID in CCD
+    Given   a user with [a detailed profile in CCD]
+    When    a request is prepared with appropriate values
+    And     the request [provide an invalid Case type ID in CCD]
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/F-049.feature]: Grammar please please please

File: src/aat/resources/features/F-049/S-233.td.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
+{
+  "_guid_": "S-233",
+  "_extends_": "F-049_Test_Data_Base",
+  "title": "must return 403 when request provide invalid Idam user ID in CCD",
+  "specs": [
+    "a detailed profile in CCD",
+    "Start case creation as Case worker",
+    "CCD Data Store",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-233.td.json]: Why are these here?

File: src/aat/resources/features/F-049/S-511.td.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
+{
+  "_guid_": "S-511",
+  "_extends_": "F-049_Test_Data_Base",
+  "title": "must return 403 when request provide invalid Jurisdiction ID in CCD",
+  "specs": [
+    "a detailed profile in CCD",
+    "Start case creation as Case worker",
+    "CCD Data Store",
```
Comment (MSancaktutar) [inline src/aat/resources/features/F-049/S-511.td.json]: These are places in their own attributes in test data object

File: src/aat/resources/features/F-049/F-049.feature (hunk 1 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/F-049.feature]: changed as suggested

File: src/aat/resources/features/F-049/F-049.feature (hunk 5 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/F-049.feature]: as per your previous comment had changed 
i.e. we shouldn't name all bad inputs as "invalid". that is vague wording. if the jurisdiction id is not available in ccd for this test, we should expressly describe it like "a jurisdiction id not existing in ccd". there are many places like this, which doesn't tell why a bad input is "invalid".

File: src/aat/resources/features/F-049/F-049.feature (hunk 7 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/F-049.feature]: as per your previous comment had changed i.e. 
please please use as proper a grammar as possible: here, this would require something like [provides an invalid Case Type ID]

File: src/aat/resources/features/F-049/S-232.td.json (hunk 4 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-232.td.json]: as we are not allowed to change the user name and password hence changing the JID which is not accessible to the user, is it make sense please let me know if some other option we have

File: src/aat/resources/features/F-049/S-233.td.json (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-233.td.json]: as  "Start case creation as Case worker", and "CCD Data Store" can be a part of base.td.json  but due to specs is an array we can put into base.td.json  or if you can suggest some other option will change

File: src/aat/resources/features/F-049/S-511.td.json (hunk 9 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-511.td.json]: as "Start case creation as Case worker", and "CCD Data Store" can be a part of base.td.json but due to specs is an array we can put into base.td.json or if you can suggest some other option will change

File: src/aat/resources/features/F-049/S-232.td.json (hunk 3 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-232.td.json]: as the above is an array we cant move some part to base data, framework dependency

File: src/aat/resources/features/F-049/S-232.td.json (hunk 4 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-232.td.json]: as we can't change the user name and password hence changing the **unauthorized jurisdiction id **

File: src/aat/resources/features/F-049/S-233.td.json (hunk 8 above)
Comment (shrivsksavit) [reply src/aat/resources/features/F-049/S-233.td.json]: framework dependency

File: src/aat/resources/features/F-049/S-232.td.json (hunk 3 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-049/S-232.td.json]: that wasn't the point

File: src/aat/resources/features/F-049/S-232.td.json (hunk 4 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-049/S-232.td.json]: scenario title and content are inconsistent, then.

File: src/aat/resources/features/F-049/S-233.td.json (hunk 8 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-049/S-233.td.json]: not really.

File: src/aat/resources/features/F-049/S-511.td.json (hunk 9 above)
Comment (MSancaktutar) [reply src/aat/resources/features/F-049/S-511.td.json]: anyway.. fixed myself.
